### ISSUE TITLE
Few "problems" with the new version

### ISSUE THREAD

**Issue by TarisGN** [TanguyOrtegat/esx_jb_eden_garage2#47]
- We can't park a vehicle into another garage, always need to pay to transfer. (maybe it's "on purpose")
- Sometimes, when you have multiple vehicles on garage and you transfer several of them, you transfer the 1st, than when you wanna do the second, it bring back the 1st one (take the money and don't transfer the second until you do it again)
- It also happen that when you have several vehicle on garage, that you chose ont of them, it's another one that get out of garage.
- When you transfer to another garage(1) then another one(2) after, you pop on the past garage(1) with the vehicle.

**Comment by TanguyOrtegat** [comment]
hey, thanks for the "issues"
normally you can park free your vehicles wherever you want, but if you parked in noth garage and you want to get out in south you will have to pay fee.
i ll try when i got the time the other issues

**Comment by TanguyOrtegat** [comment]
i just heard a bug from my players if you put vehicle out and go to other gaage and want to take out same vehicle it teleports you to other garage. will have a look into it

**Comment by TarisGN** [comment]
> hey, thanks for the "issues"
> normally you can park free your vehicles wherever you want, but if you parked in noth garage and you want to get out in south you will have to pay fee.
> i ll try when i got the time the other issues

Can it be related to database ? because for me, players can park vehicles on another garage, but they have to wait for a reboot to have it for "free", by default, if they park, garage ask to pay the fee

**Comment by JogiTE** [comment]
I can confirm that we have the same problems. Garage location is usually only after the second time parked in the same garage in the database. And that with teleporting happens here as well.

**Comment by TanguyOrtegat** [comment]
thanks for reporting! will try to fix it as soon as possible but going to vacation so big chance will be afterward.

**Comment by Alevhovic** [comment]
Hi @TanguyOrtegat 

Your update is really good, but as @JogiTE say. We have some "new" issue with it.

Just a little question : What "vehiclelock" script are you using on your server ?

**Comment by TanguyOrtegat** [comment]
@Alevhovic  i am using my own vehicle lock script. issues  will be debugged after 12 september. when i come back from vacation.

**Comment by maguhd** [comment]
I have the same issues as the guys above me :) Waiting for the fix. Have a nice vacation!!!

**Comment by madalton** [comment]
bonsoir,

j ai print un peu partout avec `print(vehicleProps)` pour comprendre a quel moment le script passait a l'ancien vehicule et c est en passant dans ce bout de code. si cela peux aider, j ai fais plein de modif sans succes.... je miserais sur elements = elem qui efface le nouveau vehicule precedement selectionne ?

`
ESX.UI.Menu.Open(
								'default', GetCurrentResourceName(), 'transfer_menu',
								{
								css =  'vehicle',
									title    =  "Voulez vous transférer: "..data.current.vehicleName.." a votre garage?",
									align    = 'top-left',
									elements = elem,
								},
`

bon courage ! et encore merci j adore votre travail.

**Comment by boomiiee** [comment]
its not fixed the second car still spawns the first car. And everyone keeps teleporting across the map when they transferred the car from one place to another.

**Comment by iTzCrutchie** [comment]
Just manually pushed this update to my scripts as i have translated everything to english and didnt want to lose the translation..

Anyways it appears to be working on my end. Transfered the first vehicle to an ew garage stored it then went back and pulled my second vehicle out no problem

Alos it is stating i can not pull out the vehicle again after i have already transfered/pull it out at the new garage. And not teleporting across the map. Though i didnt have that issue to begin with.

**Comment by madalton** [comment]
Hi,

when i want transfered 2 car and if first car is not stored and i call second car, garage spawn my first car.

**Comment by TanguyOrtegat** [comment]
should be fixed by now

**Comment by madalton** [comment]
hi,


yes is perfect. Thx

**Comment by iTzCrutchie** [comment]
So I am teleporting across the map when pulling out vehicles. It appears that with the first time you transfer a vehicle to a new garage after storing it in a different garage it will teleport you to the garage you originally stored the vehicle in. Once it teleports you if you go back to the garage where you initiated the transfer to begin with and store the vehicle you will not telpeort when you pull it back out.

**Comment by TanguyOrtegat** [comment]
@iTzCrutchie never had that problem on my end.. but could be an issue. Could you explain step by step what you did so i can try myself and test it out and debug if it happens :)

**Comment by iTzCrutchie** [comment]
@TanguyOrtegat Sure thing.
1. Transfer the vehicle to a new garage. ( I was using Pillbox Garage)
2. Vehicle was transferred out and spawned in Pillbox normally
3. Stored vehicle in Pillbox Garage
4. Went to Grove street garage and transferred the vehicle to grove street
5. Transfer was successful but teleported to Pillbox

It only happened twice but wasn't able to replicate it after that.

**Comment by TanguyOrtegat** [comment]
> @TanguyOrtegat Sure thing.
> 
> 1. Transfer the vehicle to a new garage. ( I was using Pillbox Garage)
> 2. Vehicle was transferred out and spawned in Pillbox normally
> 3. Stored vehicle in Pillbox Garage
> 4. Went to Grove street garage and transferred the vehicle to grove street
> 5. Transfer was successful but teleported to Pillbox
> 
> It only happened twice but wasn't able to replicate it after that.

hey, thanks for detailed debug, but tried many times, still not teleported ...

**Comment by iTzCrutchie** [comment]
I havent teleported either. I believe we are all good now. Thanks for the quick update :)

**Comment by TanguyOrtegat** [comment]
closing the issue
